### PR TITLE
Added cycle_sort.c

### PR DESCRIPTION
**Pull request by YashovardhanSiramdas**
**Changes:**
Added Cycle Sort in C.

### CODE REVIEW

File: code/sorting/src/cycle_sort/cycle_sort.c [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ 
+        if (pos != cycle_start)
+            swap(&item, &arr[pos]);
+ 
+        while (pos != cycle_start) {
+            pos = cycle_start;
+ 
+            for (i = cycle_start + 1; i < n; i++)
+                if (arr[i] < item)
+                    pos += 1;
```
Comment (arnavb) [inline code/sorting/src/cycle_sort/cycle_sort.c]: ++?

File: code/sorting/src/cycle_sort/cycle_sort.c [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        int i;
+        for (i = cycle_start + 1; i < n; i++)
+            if (arr[i] < item)
+                pos++;
+
+        if (pos == cycle_start)
+            continue;
+ 
+        while (item == arr[pos])
+            pos += 1;
```
Comment (arnavb) [inline code/sorting/src/cycle_sort/cycle_sort.c]: ++?

File: code/sorting/src/cycle_sort/cycle_sort.c [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    int arr[n];
+    printf("Enter %d Integers:- \n", n);
+    int i;
+    for (i = 0; i < n; i++)
+        scanf("%d", &arr[i]);
+
+    cycleSort(arr, n);
+    
+    printf("Array after Sorting:- \n");
+    for (i = 0; i < n; i++)
```
Comment (arnavb) [inline code/sorting/src/cycle_sort/cycle_sort.c]: Pre-increment.

**Comment by YashovardhanSiramdas** [comment]
Made suggested changes.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**